### DISCUSSION TITLE
Naming issue of the FooThread and alike

### DISCUSSION THREAD

**Discussion by izgzhen** [servo/servo#11271] [General]
This is a follow-up of discussion started from https://github.com/servo/servo/pull/11189.

The naming issues include:
- `Thread` is not `IpcSender`: we type-alias a lot like `FooThread = IpcSender<FooMsg>`. The "thread" here might not be indicating what we want
- `net_traits::IpcSend` is coupled with `Ipc` which might not be a good name (as suggested by @nox)
- `net_traits::CoreResourceMsg` and others renamed from original `ControlMsg` and `ResourceThread`. But the `CoreResourceMsg` manages a lot of duties and used by a lot of other code, which might add difficulty to name it cleanly.

**Comment by KiChjang** [comment]
Off the top of my head, we can rename it as `FooChannel` or `FooPipe`, since it is some sort of message passing system.

**Comment by izgzhen** [comment]
+1 for the FooChannel one

**Comment by izgzhen** [comment]
BTW we might use FooChan if the longer one is too verbose.

Also the irc paste for reference here:

```
10:46 PM <ajeffrey> in fact I prefer not introducing type synonyms for IpcSender<FooMsg> in general,
10:46 PM <nox> ajeffrey: That doesn't sound great.
10:46 PM <ajeffrey> as it makes it trickier to track down the FooMsg definition.
10:46 PM <nox> We don't care whether it's IPC or not. We care about the interface IMO.
```

**Comment by Ms2ger** [comment]
-100 on "channel" or "chan". The thing you send on is a "sender"; a "channel" is something with two sides.

**Comment by KiChjang** [comment]
How about `FooMsgSender`?

**Comment by nox** [comment]
`FooSender` seems ok to me, or just not introducing type synonyms at all as @asajeffrey seemed to prefer.

**Comment by nox** [comment]
Do we still care about this?

**Comment by asajeffrey** [comment]
Yes, but it's low priority. While we're at it, the naming of the *Msg types is a bit erratic.

**Comment by nox** [comment]
The naming is starting to get on my nerves. ><
